packages/cli/tests/commonlib/functionValidator.ts

```diff
@@ -285,4 +217,37 @@ export class FunctionValidator {
       return undefined;
     }
   }
+
+  private getExpectedM365ApplicationIdUri(ctx: any, activeResourcePlugins: string[]): string {
+    let expectedM365ApplicationIdUri = "";
+    if (activeResourcePlugins.includes(PluginId.FrontendHosting)) {
+      const tabDomain = ctx[PluginId.FrontendHosting][StateConfigKey.domain];
+      const m365ClientId = ctx[PluginId.Aad][StateConfigKey.clientId];
+      expectedM365ApplicationIdUri =
+        `api://${tabDomain}/` +
+        (activeResourcePlugins.includes(PluginId.Bot)
+          ? `botid-${ctx[PluginId.Bot][StateConfigKey.botId]}`
+          : `${m365ClientId}`);
+    } else if (activeResourcePlugins.includes(PluginId.Bot)) {
+      expectedM365ApplicationIdUri = `api://botid-${ctx[PluginId.Bot][StateConfigKey.botId]}`;
+    }
+    return expectedM365ApplicationIdUri;
+  }
+
+  private async getM365ClientSecret(
+    activeResourcePlugins: string[],
+    resourceBaseName: string
+  ): Promise<string> {
+    let m365ClientSecret: string;
+    if (activeResourcePlugins.includes(PluginId.KeyVault)) {
+      m365ClientSecret = `@Microsoft.KeyVault(VaultName=${resourceBaseName};SecretName=m365ClientSecret)`;
```

Comment:
Why is this SecretName hardcode? Can we get it from keyvault's config?

Reply:
Will fix in next PR:
Pass `keyVaultName` `m365ClientSecretName` from parameter.json file.